ktor-features/ktor-thymeleaf/jvm/src/io/ktor/thymeleaf/Thymeleaf.kt

```diff
@@ -0,0 +1,86 @@
+package io.ktor.thymeleaf
+
+import io.ktor.application.ApplicationCallPipeline
+import io.ktor.application.ApplicationFeature
+import io.ktor.http.ContentType
+import io.ktor.http.charset
+import io.ktor.http.content.EntityTagVersion
+import io.ktor.http.content.OutgoingContent
+import io.ktor.http.content.versions
+import io.ktor.http.withCharset
+import io.ktor.response.ApplicationSendPipeline
+import io.ktor.util.AttributeKey
+import io.ktor.util.cio.bufferedWriter
+import kotlinx.coroutines.io.ByteWriteChannel
+import org.thymeleaf.TemplateEngine
+import org.thymeleaf.context.Context
+
+/**
+ * Represents a content handled by [Thymeleaf] feature.
+ *
+ * @param template name that is resolved by thymeleaf
+ * @param model to be passed during template rendering
+ * @param etag value for `E-Tag` header (optional)
+ * @param contentType of response (optional, `text/html` with UTF-8 character encoding by default)
+ */
+class ThymeleafContent(
+    val template: String,
+    val model: Map<String, Any>,
+    val etag: String? = null,
+    val contentType: ContentType = ContentType.Text.Html.withCharset(Charsets.UTF_8)
+)
+
+/**
+ * Thymeleaf support feature. Provides ability to respond with [Thymeleaf]
+ */
+class Thymeleaf(private val engine: TemplateEngine) {
```

Comment:
Doesn't it require/provide any configuration? For example, resolution rules?

Reply:
https://github.com/t-kameyama/ktor/blob/b726a79ed877d57ebe970383dde42eb7cd81d05b/ktor-features/ktor-thymeleaf/jvm/test/it/ktor/thymeleaf/ThymeleafTest.kt#L142-L148

Reply:
We set configuration like this:
```kotlin
application.install(Thymeleaf) {
    val resolver = ClassLoaderTemplateResolver()
    resolver.setTemplateMode("HTML")
    resolver.prefix = "templates/"
    resolver.suffix = ".html"
    setTemplateResolver(resolver)
}
```